src/main/java/org/jabref/gui/desktop/os/Linux.java

```diff
@@ -89,41 +90,62 @@ public void openFileWithApplication(String filePath, String application) throws
     public void openFolderAndSelectFile(Path filePath) throws IOException {
         String desktopSession = System.getenv("DESKTOP_SESSION");
 
-        String cmd = "xdg-open " + filePath.toAbsolutePath().getParent().toString(); // default command
+        String absoluteFilePath = filePath.toAbsolutePath().toString();
+        String[] cmd = {"xdg-open", absoluteFilePath}; // default command
 
         if (desktopSession != null) {
             desktopSession = desktopSession.toLowerCase(Locale.ROOT);
             if (desktopSession.contains("gnome")) {
-                cmd = "nautilus --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = new String[] {"nautilus", "--select", absoluteFilePath};
             } else if (desktopSession.contains("kde") || desktopSession.contains("plasma")) {
-                cmd = "dolphin --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = new String[] {"dolphin", "--select", absoluteFilePath};
             } else if (desktopSession.contains("mate")) {
-                cmd = "caja --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = new String[] {"caja", "--select", absoluteFilePath};
             } else if (desktopSession.contains("cinnamon")) {
-                cmd = "nemo --select " + filePath.toString().replace(" ", "\\ ");
+                cmd = new String[] {"nemo", absoluteFilePath}; // Although nemo is based on nautilus it does not support --select, it directly highlights the file
             }
         }
-        Runtime.getRuntime().exec(cmd);
+        ProcessBuilder processBuilder = new ProcessBuilder((cmd));
+        Process process = processBuilder.start();
+
+        StreamGobbler streamGobblerInput = new StreamGobbler(process.getInputStream(), LOGGER::debug);
+        StreamGobbler streamGobblerError = new StreamGobbler(process.getErrorStream(), LOGGER::debug);
+
+        JabRefExecutorService.INSTANCE.execute(streamGobblerInput);
+        JabRefExecutorService.INSTANCE.execute(streamGobblerError);
     }
 
     @Override
     public void openConsole(String absolutePath) throws IOException {
-        Runtime runtime = Runtime.getRuntime();
-        Process p = runtime.exec("readlink /etc/alternatives/x-terminal-emulator");
-        BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
-
-        String emulatorName = reader.readLine();
```

Comment:
I get null here. I think /etc/alternatives is a Debian thing.

Reply:
Maybe if (Files.exists(/etc/alternatives...) ?

Reply:
Yes, but what if it does not exist? What terminal emulator do we choose then?

Reply:
Turns out there is no real solution, just checking multiple places https://superuser.com/a/1461079

Reply:
So how do we proceed? Leave it as is? Show an alert message that we cannot determine a TE?
Do you think this is a feature that many people use?

Reply:
Show an alert message. Unable to open terminal. I doubt many use this